### PR TITLE
Added Devise

### PR DESCRIPTION
**Pull request by jaffemd**
Recommended way of signing in is to choose an existing user's email address and clicking on forgot password...creating a new user doesn't work right because the user model for this app is ridiculous

### CODE REVIEW

File: db/20180329014741_add_devise_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.string   :current_sign_in_ip
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
```
Comment (thomasythuang) [inline db/20180329014741_add_devise_to_users.rb]: are these just less necessary or more of a pain to implement?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,9 +1,6 @@
 class ApplicationController < ActionController::Base
   attr_accessor :repo
-
-  def current_user
-    @current_user = repo.user.find_by_id(session[:user_id]) if session[:user_id]
-  end
```
Comment (thomasythuang) [inline app/controllers/application_controller.rb]: `current_user` is used in `admin?` below...I'm going to guess that it's re-defined by `authenticate_user!`?

File: app/controllers/analytics_controller.rb [hunk 3]
```diff
@@ -1,6 +1,4 @@
 class AnalyticsController < ApplicationController
-  before_filter :authenticate, :employee?
```
Comment (thomasythuang) [inline app/controllers/analytics_controller.rb]: If nothing else uses these methods we should strip them out in a clean-up MR later to simplify the application controller

File: db/20180329014741_add_devise_to_users.rb (hunk 1 above)
Comment (jaffemd) [reply db/20180329014741_add_devise_to_users.rb]: This migration was auto-generated by devise and I just stuck with the default things (this was commented out by default). But we can always look more into what these things do and see if we need them.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (jaffemd) [reply app/controllers/application_controller.rb]: Yup, devise or rails or something comes with a `current_user` method that's much better than this one...

File: db/20180329014741_add_devise_to_users.rb (hunk 1 above)
Comment (thomasythuang) [reply db/20180329014741_add_devise_to_users.rb]: oh cool, default settings is definitely good for now

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (thomasythuang) [reply app/controllers/application_controller.rb]: awesome

File: app/controllers/analytics_controller.rb (hunk 3 above)
Comment (jaffemd) [reply app/controllers/analytics_controller.rb]: 👍 Yeah, the existing `authenticate` method should probably fully go away

**Comment by thomasythuang** [comment]
Is there a specific order to be following for the setup tasks?

I pulled down the branch and got an error on `bundle exec rake db:setup`

```
-----------------------Adding Users-------------------------
rake aborted!
Validation failed: Email is invalid, Password can't be blank
/Users/thomas.huang/.gem/ruby/2.3.1/gems/activerecord-4.1.0/lib/active_record/validations.rb:57:in `save!'
```

**Pull request merged**